subprojects/gradle/test-project/src/main/kotlin/com/avito/test/gradle/dependencies/GradleDependency.kt

```diff
@@ -0,0 +1,75 @@
+package com.avito.test.gradle.dependencies
+
+import com.avito.test.gradle.dependencies.GradleDependency.Coordinate.Maven
+import com.avito.test.gradle.dependencies.GradleDependency.Coordinate.Platform
+import com.avito.test.gradle.dependencies.GradleDependency.Coordinate.Project
+import org.funktionale.either.Either
+import org.gradle.util.Path
+
+data class GradleDependency(
```

Comment:
With all respect to intention (hate dynamic things), i honestly don't like the final result, and that's why:

1. i need to learn another api around gradle that is less configurable and can possibly block some uncommon things if future
2. strings was chosen exactly because of this ^^ , as lesser evil
3. new api is misleading: what is maven? afaik it's artifacts coordinates notation, introduced by gradle, compatible with maven/ivy etc

Reply:
+ for maven

Reply:
> i need to learn another api around gradle that is less configurable
Don't agree that we need "super all cases Gradle API". Only we need is a simple type-safe way to declare dependencies for all our cases. I prefer cut API to String parameter
> strings was chosen exactly because of this ^^ , as lesser evil
I don't like String because I spent time digging into the TestProjectGenerator implementation to realize how it actually will be pasted to `build.gradle`
> new api is misleading: what is maven?
that only point I agree with. Could you help me choose correct name?

Reply:
Possible option: external dependency (https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/dsl/DependencyHandler.html)